src/main/java/com/google/sps/servlets/BlobstoreUploadUrlServlet.java

```diff
@@ -0,0 +1,40 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * When the fetch() function requests the /blobstore-upload-url URL, the content of the response is
+ * the URL that allows a user to upload a file to Blobstore.
```

Comment:
This is an interesting idea, and I like how you documented it. But is the idea that the javascript code will retrieve a URL this way and then interact directly with Blobstore? It seems unsafe to permit the front-end to do this.

This is exactly the sort of decision that https://overthewire.org/wargames/natas will help you make.

Reply:
I got the logic of this function from the other intern who had figured out how to work around the blobstore errors. From my understanding of the function, it's creating a url for the image that was uploaded and giving that back to the javascript (so changes in the javascript wouldn't create any points of contact between the javascript and the blobstore service). That said, I could be understanding it wrong and we haven't started playing the game yet. 
Is there a threat I'm missing?

Reply:
The doc says the response contains a "URL that allows a user to *upload* a file to blobstore". Whereas I think you're saying the returned URL can be used to *download* the file.

It looks to me like the former is true. In which case the user is able to upload whatever they want to blobstore.